File: SkaiLab.Invoice/ClientApp/src/app/models/tax-monthly.ts
import { Currency } from "./currency";
import { Invoice } from "./customer-transaction";
import { Expense } from "./expense";
import { PayrollMonthTax } from "./payroll";

export class TaxMonthly{
    totalInvoice:number;
    totalExpense:number;
    totalEmployeeSalary:number;
    totalPayToTax:number;
    totalPayToTaxInKHR:number;
    isPayrollRun:boolean;
    invoices:Invoice[];
    expenses:Expense[];
    payroll:PayrollMonthTax;
    currency:Currency;
    taxCurrency:Currency;
    totalInvoiceTaxInBaseCurrency:number;
    totalInvoiceTaxInTaxCurrency:number;
    totalExpenseTaxInBaseCurrency:number;
    totalExpenseTaxInTaxCurrency:number;
    totalEmployeeTaxInBaseCurrency:number;
    totalEmployeeTaxInTaxCurrency:number;
    showInvoicePanel:boolean;
    showExpensePanel:boolean;
    showEmployeePanel:boolean;
    constructor(){
        this.totalInvoice=0;
        this.totalExpense=0;
        this.totalEmployeeSalary=0;
        this.totalPayToTax=0;
        this.totalPayToTaxInKHR=0;
        this.isPayrollRun=true;
        this.invoices=[];
        this.expenses=[];
        this.payroll=new PayrollMonthTax();
        this.currency=new Currency();
        this.taxCurrency=new Currency();
        this.totalInvoiceTaxInBaseCurrency=0;
        this.totalInvoiceTaxInTaxCurrency=0;
        this.totalExpenseTaxInBaseCurrency=0;
        this.totalExpenseTaxInTaxCurrency=0;
        this.totalEmployeeTaxInBaseCurrency=0;
        this.totalEmployeeTaxInTaxCurrency=0;
    }

}

File: SkaiLab.Invoice.Models/Report/TaxMonthly.cs
using System.Collections.Generic;
namespace SkaiLab.Invoice.Models.Report
{
    public class TaxMonthly
    {
        public decimal TotalInvoice { get; set; }
        public decimal TotalExpense { get; set; }
        public decimal TotalEmployeeSalary { get; set; }
        public decimal TotalPayToTax { get; set; }
        public decimal TotalPayToTaxInKHR { get; set; }
        public bool IsPayrollRun { get; set; }
        public List<Invoice> Invoices { get; set; }
        public List<Expense> Expenses { get; set; }
        public PayrollMonthTax Payroll { get; set; }
        public Currency Currency { get; set; }
        public Currency TaxCurrency { get; set; }
        public decimal TotalInvoiceTaxInBaseCurrency { get; set; }
        public decimal TotalInvoiceTaxInTaxCurrency { get; set; }
        public decimal TotalExpenseTaxInBaseCurrency { get; set; }
        public decimal TotalExpenseTaxInTaxCurrency { get; set; }
        public decimal TotalEmployeeTaxInBaseCurrency { get; set; }
        public decimal TotalEmployeeTaxInTaxCurrency { get; set; }
    }
}
